scripts/make-g-cloud-7-live.py

```diff
@@ -0,0 +1,148 @@
+"""
+
+Usage:
+    scripts/make-g-cloud-7-live.py <data_api_url> <data_api_token> <draft_bucket> <documents_bucket> [--dry-run]
+"""
+import sys
+sys.path.insert(0, '.')
+
+try:
+    import urlparse
+except ImportError:
+    import urllib.parse as urlparse
+
+from itertools import islice
+import random
+import re
+
+from docopt import docopt
+from dmutils.apiclient import DataAPIClient
+from dmutils.s3 import S3
+
+DOCUMENT_KEYS = [
+    'pricingDocumentURL', 'serviceDefinitionDocumentURL',
+    'sfiaRateDocumentURL', 'termsAndConditionsDocumentURL',
+]
+FRAMEWORK_SLUG = 'g-cloud-7'
+
+
+def assert_equal(one, two):
+    if one != two:
+        raise AssertionError("{} != {}".format(one, two))
+
+
+def find_suppliers_on_framework(client, framework_slug):
+    return (
+        supplier for supplier in client.find_framework_suppliers(FRAMEWORK_SLUG)['supplierFrameworks']
+        if supplier['onFramework']
+    )
+
+
+def find_submitted_draft_services(client, supplier, framework_slug):
+    return (
+        draft_service for draft_service in
+        client.find_draft_services(supplier['supplierId'], framework=framework_slug)['services']
+        if draft_service['status'] == 'submitted' and not draft_service.get('serviceId')
+    )
+
+
+def parse_document_url(url, framework_slug):
+    pattern = r'/{}/(\d+)/(\d+)-(.*)$'.format(re.escape(framework_slug))
+    match = re.search(pattern, url)
+    if not match:
+        raise ValueError("Could not parse document URL {}".format(url))
+    return {
+        'domain': urlparse.urlparse(url).netloc,
+        'supplier_id': match.group(1),
+        'draft_id': match.group(2),
+        'document_name': match.group(3),
+    }
+
+
+def get_draft_document_path(parsed_document, framework_slug):
+    return '{framework_slug}/{supplier_id}/{draft_id}-{document_name}'.format(
+        framework_slug=framework_slug,
+        **parsed_document)
+
+
+def get_live_document_path(parsed_document, framework_slug, service_id):
+    return '{framework_slug}/documents/{supplier_id}/{service_id}-{document_name}'.format(
+        framework_slug=framework_slug,
+        service_id=service_id,
+        **parsed_document)
+
+
+def get_live_asset_domain(domain):
+    if domain.startswith('preview'):
+        return 'preview-assets.development.digitalmarketplace.service.gov.uk'
+    elif domain.startswith('staging'):
+        return 'staging-assets.digitalmarketplace.service.gov.uk'
+    elif domain.startswith('www'):
+        return 'assets.digitalmarketplace.service.gov.uk'
+
+
+def get_live_asset_url(parsed_document, live_document_path):
+    domain = get_live_asset_domain(parsed_document['domain'])
+    return 'https://{}/{}'.format(domain, live_document_path)
+
+
+def make_draft_service_live(client, supplier, draft_service, framework_slug):
+    print("  > Migrating draft {} - {}".format(draft_service['id'], draft_service['serviceName']))
```

Comment:
We could use the script logging helper instead of `print` https://github.com/alphagov/digitalmarketplace-scripts/blob/master/dmscripts/logging.py. This would also enable the API client logging, which could be useful